pipeline.py

```diff
@@ -1,20 +1,110 @@
-    pass
+    possible_players_four = []
+    possible_players_five = []
+    possible_players_six = []
+    # Add player to the set of the of times that they can play in
+    for index, player in input_df.iterrows():
+        possible_players_four, possible_players_five, possible_players_six = append_time_group(player[SID], player[FIRST_PREF], possible_players_four, possible_players_five, possible_players_six)
+        # If they signed up with a second prefernce, also add them to the list of players for the second time preference
+        if(player[SECOND_PREF]):
+            possible_players_four, possible_players_five, possible_players_six = append_time_group(player[SID], player[SECOND_PREF], possible_players_four, possible_players_five, possible_players_six)
+
+    seen = set()
+    output = dict()
+    # Fill up the time slots with the least number of players who can play in that slot first
+    # hahahahaha sphagetti code
+    if len(possible_players_four) <= len(possible_players_five) and len(possible_players_four) <= len(possible_players_six):
+        seen, four = add_new_players(seen, possible_players_four)
+        output[FOUR] = four
+        if len(possible_players_five) <= len(possible_players_six):
+            seen, five = add_new_players(seen, possible_players_five)
+            output[FIVE] = five
+            seen, six = add_new_players(seen, possible_players_six)
+            output[SIX] = six
+        else:
+            seen, six = add_new_players(seen, possible_players_six)
+            output[SIX] = six
+            seen, five = add_new_players(seen, possible_players_five)
+            output[FIVE] = five
+    elif len(possible_players_five) <= len(possible_players_four) and len(possible_players_five) <= len(possible_players_six):
+        seen, five = add_new_players(seen, possible_players_five)
+        output[FIVE] = five
+        if len(possible_players_four) <= len(possible_players_six):
+            seen, four = add_new_players(seen, possible_players_four)
+            output[FOUR] = four
+            seen, six = add_new_players(seen, possible_players_six)
+            output[SIX] = six
+        else:
+            seen, six = add_new_players(seen, possible_players_six)
+            output[SIX] = six
+            seen, four = add_new_players(seen, possible_players_four)
+            output[FOUR] = four
+    else:
+        seen, six = add_new_players(seen, possible_players_six)
+        output[SIX] = six
+        if len(possible_players_four) <= len(possible_players_five):
+            seen, four = add_new_players(seen, possible_players_four)
+            output[FOUR] = four
+            seen, five = add_new_players(seen, possible_players_five)
+            output[FIVE] = five
+        else:
+            seen, five = add_new_players(seen, possible_players_five)
+            output[FIVE] = five
+            seen, four = add_new_players(seen, possible_players_four)
+            output[FOUR] = four
```

Comment:
we also need to put in the waitlist object into dataframe so that people know they arewaitlisted